src/components/ItemRecognition/ItemRecognition.js

```diff
@@ -61,45 +61,60 @@ class ItemRecognition extends Component {
     );
   };
 
+  setSuggestions = (items, index) => {
+    const suggestions = [];
+    while (suggestions.length < 3) {
+      if (items[index].id === 'unknown') continue;
+      suggestions.push(this.props.storeList[items[index].id]);
+      index++;
+    }
+    this.props.setSuggestions(suggestions);
+  };
+
   handleImg = img => {
     if (this.success) return;
+
     this.model.predict(img).then(async items => {
       const item = items[0];
       if (
-        (item.value > ML_THRESHOLD &&
-          item.id !== 'unknown' &&
-          !this.props.prediction) ||
-        (Date.now() - this.scanningStartTime) / 1000 > TIMEOUT_IN_SECONDS
+        item.value > ML_THRESHOLD &&
+        item.id !== 'unknown' &&
+        !this.props.prediction
       ) {
+        // Item recognised
         this.success = true;
         this.addTrainingImage(img.src, item.id);
+        this.setSuggestions(items, 1);
         await this.props.setPrediction(item.id, img.src);
-        const suggestions = [];
-        for (let i = 1; i < 4; i++) {
-          suggestions.push(this.props.storeList[items[i].id]);
-        }
-        this.props.setSuggestions(suggestions);
+
         this.webcam.current.success(() => {
           this.setState({text: 'Snack recognised!', subText: null});
           setTimeout(() => {
-            this.props.history.replace(
-              item.id === 'unknown' ? '/editsnack' : '/confirmitem'
-            );
+            this.props.history.replace('/confirmitem');
           }, 500);
         });
-      } else {
-        if (
-          !this.state.subText &&
-          (Date.now() - this.scanningStartTime) / 1000 >
-            TIMEOUT_IN_SECONDS - SHOW_RETRY_FOR
-        )
-          this.setState({
-            text: "We can't recognise the snack",
-            subText: 'Try turning the snack so the logo is seen by the camera'
-          });
-        if (this.webcam.current)
-          this.webcam.current.requestScreenshot().then(this.handleImg);
+      } else if (
+        (Date.now() - this.scanningStartTime) / 1000 >
```

Comment:
Can we extract the date checks to a single variable?

```
const hasTimedOut = (Date.now() - this.scanningStartTime) / 1000 > TIMEOUT_IN_SECONDS

const showRotationMessage = !this.state.subText && (Date.now() - this.scanningStartTime) / 1000 > TIMEOUT_IN_SECONDS - SHOW_RETRY_FOR

if (hasTimedOut) {
  ...
} else if (showRotationMessage) {
  this.setState({
    ...
  })
}
```

We could probably go one step further and create a helper function

```
hasBeen(seconds) {
  return (Date.now() - this.scanningStartTime) / 1000) > seconds
}

const hasTimedOut = this.hasBeen(TIMEOUT_IN_SECONDS)
const showRotationMessage = !this.state.subText && this.hasBeen(TIMEOUT_IN_SECONDS - SHOW_RETRY_FOR)
```